platform-api-client-gwt/che-core-client-gwt-project/src/main/java/org/eclipse/che/api/project/gwt/client/ProjectServiceClient.java

```diff
@@ -147,6 +147,18 @@ void cloneProjectToCurrentWorkspace(String srcWorkspaceId,
      */
     void resolveSources(String workspaceId, String path, AsyncRequestCallback<List<SourceEstimation>> callback);
 
+     /**
+      * Gets list of {@link SourceEstimation} for all supposed project types.
+      *
+      * @param workspaceId
+      *         id of current workspace
+      * @param path
+      *         path of the project to resolve
+      * @return a promise that will provide a list of {@code SourceEstimation} for the given {@code workspaceId} and {@code path},
+      *         or rejects with on error
+      */
+    Promise<List<SourceEstimation>> resolveSources(String workspaceId, String path);
```

Comment:
I think it would be nice to mark old methods with AsyncRequestCallback as Deprecated